ibis/expr/scope.py

```diff
@@ -41,19 +41,63 @@
 from ibis.expr.timecontext import TimeContextRelation, compare_timecontext
 from ibis.expr.typing import TimeContext
 
-ScopeItem = namedtuple('item', ['value', 'timecontext'])
+ScopeItem = namedtuple('item', ['timecontext', 'value'])
 
 
 class Scope:
-    def __init__(self, items: Dict[str, ScopeItem] = None):
-        self._items = items or {}
+    def __init__(
```

Comment:
This is the change to add a constructor for Scope to take a dict as init param @icexelloss

Reply:
I see.

I found this constructor to a little intuitive. Why does this takes mapping from op -> result but with only one timecontext?

Reply:
Let me explain about my consideration here:
This API is aimed to serve those scope initializations with a dict of (op, value). We have several usages (mostly param) here and there. No time context is provided in those params, and they share **the same time context** (defined by expression).  There is no usage that init a Scope with a dict of (op, value), but with different time context for each item in dict. 

I kind of resist implementing this API that takes `Dict[str, ScopeItem]` because for the usage I mentioned above, will have to deconstruct the original dict and wrap time context in it, make ScopeItem and form a new dict, then call this constructor.

Reply:
In our execution phase, `(op, result)` is generated one by one and every pari has a `timecontext` associated, so they could be wrap up as a `Scope` and we will call `merge_scope` on it.  There is no such a intermediate result that gives us a dict of `(op, result)`, except for those params in input.

Reply:
Ok, not a big deal. A little strange but perhaps more convenient this way.